src/field_theory/finite.lean

```diff
@@ -9,15 +9,40 @@ import data.zmod.basic
 import linear_algebra.basis
 import ring_theory.integral_domain
 
-universes u v
-variables {K : Type u} {R : Type v}
+/-!
+# Finite fields
 
-open function finset polynomial nat
+This file contains basic results about finite fields.
+Throughout most of this file, `K` denotes a finite field with `q` elements.
+
+## Main results
+
+1. Every finite integral domain is a field (`field_of_integral_domain`).
+2. The unit group of a finite field is a cyclic group of order `q - 1`.
+   (`finite_field.is_cyclic` and `card_units`)
+3. `sum_pow_units`: The sum of `x^i`, where `x` ranges over the units of `K`, is
+   - `q-1` if `q-1 ∣ i`
+   - `0`   otherwise
+4. `finite_field.card`: The cardinality `q` is a power of the characteristic of `K`.
+   See `card'` for a variant.
+-/
+
+variables {K : Type*} [field K] [fintype K]
+variables {R : Type*} [integral_domain R]
+local notation `q` := fintype.card K
```

Comment:
Can you mention this `local notation` in the module docstring? I overlooked it at first and then got confused later. Local notation that depends on `variables` is especially disorienting if you're not expecting it.

Reply:
It's already there, but I also added it to the `## Notation` section.